data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -43,10 +69,35 @@ public GrokPrepper(final PluginSetting pluginSetting) {
      *
      * @param records Input records that will be modified/processed
      * @return Record  modified output records
-    */
+     */
     @Override
     public Collection<Record<String>> doExecute(Collection<Record<String>> records) {
-        return records;
+        final List<Record<String>> recordsOut = new LinkedList<>();
```

Comment:
`LinkedList` is more efficient here since the only operation being done on `recordsOut` is appending to the end, and I don't think you can initialize the size.